request/errors.go

```diff
@@ -0,0 +1,29 @@
+package request
+
+import (
+	"fmt"
+	"net/http"
+)
+
+const (
+	// TODO_TECHDEBT: return the correct id field from the request
```

Comment:
This is out of scope for `request` package: it should not know about JSONRPC (or any other service spec). 
Could you please add a summary of the following to the PR, to help future readers/contributors?

```go
The responsibility of `request` package is: [Extract the service ID and find the target service's corresponding QoS instance.](https://github.com/buildwithgrove/path/blob/e0067eb0f9ab0956127c952980b09909a795b300/gateway/gateway.go#L52C2-L52C45).
Request package decides how the requested service is referenced by the user (currently: subdomain of the HTTP request). 
Processing should fail here only if:
A) It can not recognize the requested service, e.g. missing subdomain, or
B) The requested service is not found/configured for the gateway instance.

If things fail at this point: need to return a 404, because we do not understand what service the user is requesting.
```

Reply:
Added comment to `parser.go`